### PR TITLE
Add Initial support for RTL layouts

### PR DESCRIPTION
**Pull request by hadicharara**
**Motivation**

RTL languages such as Arabic require RTL layouts for a proper user experience. While usable, the current state of affairs is sub-par for users of Jellyfin in RTL languages.

**Examples**
Screenshot 2022-07-02 at 15-56-37 Jellyfin
Screenshot 2022-07-02 at 16-00-06 Jellyfin
Screenshot 2022-07-02 at 16-00-45 Jellyfin

**Philosophy**
- My changes are meant to be as unobtrusive as possible to core Jellyfin contributors. All CSS changes made will only affect users of RTL languages. When modifying the css, Jellyfin needn't worry about RTL layouts (although it is very easy to accomodate for them).
- Jellyfin already has language settings, independent of the browser's settings. I aimed to respect the Jellyfin user's language and locale settings.

**Changes**
- When globalize detects a RTL language, it injects the `dir="rtl"` attribute to the HTML body. Most of the work is done by this simple fix. 
- CSS changes are made primarily with the inline and block variant attributes for inser, margin and padding, else by checking the dir attribute.
- Javascript changes has been made to fix the emby-slider component.
- Some changes made to support alternative numeral systems (my motivation was eastern arabic numerals). This is done by using toLocaleString() and similar functions. 
- rtl.scss file added to flip horizontally directional icons. 

**Resuts**
Screenshot 2022-07-02 at 16-13-36 Jellyfin
Screenshot 2022-07-02 at 16-14-38 Jellyfin
Screenshot 2022-07-02 at 16-16-04 Jellyfin
Screenshot 2022-07-02 at 16-17-07 Jellyfin

There are probably many things that could be improved with my changes (probably some things with the usage of toLocaleString() could be improved).
I welcome any feedback!

TODO: testing on TV and mobile layouts. Thoroughly checking the application for inconsistencies in RTL layouts. No changes should be visible to LTR layout users, though this should be checked.

### CODE REVIEW

**Comment by hadicharara** [comment]
Also the mainDrawer still opens on the left. This is to be fixed (probably by changing the lib file itself).

**Comment by hadicharara** [comment]
Jellyfin has been tested in desktop, mobile and tv layouts, both in LTR and RTL layouts. All bugs I could fine have been crushed (except the fact that the dashboard drawer animation comes from the left, but that's fine). It is currently 100% usable and I have been using it for days. I await feedback!

**Comment by hadicharara** [comment]
Information for reviewers: Some elements have been hardcoded as ltr:
- The loading spinner because it should look the same in ltr and rtl.
- The video player is LTR otherwise the subtitle canvas get messed up.
- File paths are always "ltr".
- Player controls are "ltr" because it doesn't represent directionality rather the direction of a tape recording.

File: src/assets/css/librarybrowser.scss [hunk 1]
```diff
@@ -208,15 +246,16 @@
...
+    padding-inline: 2.4em 0 !important;
     flex-grow: 1;
     font-weight: 400 !important;
     margin: 0 !important;
     border-radius: 0 !important;
 }
 
 .navMenuOptionIcon {
-    margin-right: 1.2em;
+    margin-inline-end: 1.2em;
```
Comment (thornbill) [inline src/assets/css/librarybrowser.scss:258]: Unfortunately, I believe some of these newer CSS properties will cause issues on the old browsers shipped on TVs (webOS and Tizen specifically). Someone will need to test those.

Comment (hadicharara) [reply src/assets/css/librarybrowser.scss:258]: Indeed. I'll replace them with appropriate [dir="rtl"] selectors.

Comment (hadicharara) [reply src/assets/css/librarybrowser.scss]: Fixed

**Comment by CAPRICORNX10X** [comment]
hi 
you can please add the Hebrew language to the display from right to left
as in the Arabic language

**Comment by hadicharara** [comment]
This should have added support for all RTL languages in Jellyfin. However, this PR should be closed for a lack of interest, since it was't properly reviewed.

**Comment by thornbill** [comment]
Oh we are definitely interested in landing this! The problem is it's basically only me reviewing larger changes right now and I've been focusing on getting the new website going. That said I'm back around now working through reviews so if you could rebase this I'll work on reviewing it in the next few days. 💜

**Comment by hadicharara** [comment]
Alright no problem! It should be rebased soon.

**Comment by CAPRICORNX10X** [comment]
I can help you.  in the feedbacks and check that everything is correct.  And what not to update you.
________________________________

**Comment by hadicharara** [comment]
Just merged latest changes from master and tested.

**Comment by hadicharara** [comment]
Sorry. Should be good now.

**Comment by hadicharara** [comment]
Fixed all the issues you pointed out. Sorry for the delay, I was a bit busy.

File: src/scripts/globalize.js [hunk 2]
```diff
@@ -38,6 +40,43 @@ import { currentSettings as userSettings } from './settings/userSettings';
         return fallbackCulture;
     }
 
+    export function getIsRTL() {
+        return isRTL;
+    }
+
+    function checkAndProcessDir(culture) {
+        isRTL = false;
+        for (const lang of RTL_LANGS) {
```
Comment (thornbill) [inline src/scripts/globalize.js:49]: I think for `RTL_LANGS` we should use the full culture string to simplify  this check and avoid this looping. Then the check could just be `RTL_LANGS.includes(culture)`.

**Review by thornbill — CHANGES_REQUESTED** [review]

File: src/components/metadataEditor/style.scss [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    [dir="ltr"] & {
+        margin-left: auto;
+    }
+
+    [dir="rtl"] & {
+        margin-right: auto;
+    }
+}
+
+.marginLeft {
```
[the remark this answers is not in the document]
Comment (hadicharara) [reply src/components/metadataEditor/style.scss:11]: I nested it inside .metadataFormFields as there is no parent element having the dialogHeader class.

File: src/scripts/globalize.js (hunk 2 above)
Comment (hadicharara) [reply src/scripts/globalize.js:49]: The problem is that the culture parameter is sometimes of the format language, as in en or ar or pe, and sometimes is of the format en_US, ar_SA etc. Also checking for language is easier than checking for language + locale combinations. For arabic, there are 28 different locales.

File: src/scripts/globalize.js [hunk 4]
```diff
@@ -38,6 +40,43 @@ import { currentSettings as userSettings } from './settings/userSettings';
...
+    function processIsRTL() {
+        document.getElementsByTagName('body')[0].setAttribute('dir', 'rtl');
+        document.getElementsByTagName('html')[0].setAttribute('dir', 'rtl');
+        import('../styles/rtl.scss');
+    }
+
+    function processIsLTR() {
+        document.getElementsByTagName('body')[0].setAttribute('dir', 'ltr');
+        document.getElementsByTagName('html')[0].setAttribute('dir', 'ltr');
+    }
```
[the remark this answers is not in the document]
Comment (hadicharara) [reply src/scripts/globalize.js]: Done

File: src/scripts/globalize.js [hunk 5]
```diff
@@ -38,6 +40,43 @@ import { currentSettings as userSettings } from './settings/userSettings';
...
+
+    export function getElementIsRTL(element) {
```
[the remark this answers is not in the document]
Comment (hadicharara) [reply src/scripts/globalize.js]: Done

File: src/libraries/navdrawer/navdrawer.js [hunk 6]
```diff
@@ -193,7 +206,7 @@ class NavDrawer {
 
         options.target.classList.add('touch-menu-la');
         options.target.style.width = options.width + 'px';
-        options.target.style.left = -options.width + 'px';
+        options.target.style.insetInlineStart = -options.width + 'px';
```
[the remark this answers is not in the document]
Comment (hadicharara) [reply src/libraries/navdrawer/navdrawer.js]: Done

File: src/elements/emby-scrollbuttons/emby-scrollbuttons.js [hunk 7]
```diff
@@ -42,6 +42,11 @@ const EmbyScrollButtonsPrototype = Object.create(HTMLDivElement.prototype);
     function updateScrollButtons(scrollButtons, scrollSize, scrollPos, scrollWidth) {
         // TODO: Check if hack is really needed
         // hack alert add twenty for rounding errors
```
[the remark this answers is not in the document]
Comment (hadicharara) [reply src/elements/emby-scrollbuttons/emby-scrollbuttons.js]: Done

**Review by thornbill — APPROVED** [review]

**Pull request merged**